environment.py

```diff
@@ -0,0 +1,150 @@
+import os
+
+import time
+from typing import List, Tuple
+
+from color_refinement import get_number_automorphisms, is_isomorphisms
+from graph import Graph
+from graph_io import load_graph
+
+GRAPHS = 'graphs'
+BRANCHING = os.path.join(GRAPHS, 'branching')
+COLORREF = os.path.join(GRAPHS, 'colorref')
+TREEPATHS = os.path.join(GRAPHS, 'treepaths')
+
+FILE = os.path.join(COLORREF, 'colorref_smallexample_4_16.grl')
+
+
+def main():
+    if FILE == '':
+        graph_files = find_graphs()
+    else:
+        graph_files = [FILE]
+    process_graph_files(graph_files)
+
+
+def find_graphs() -> List[str]:
+    branching_graphs = [BRANCHING + file for file in os.listdir(BRANCHING)]
+    colorref_graphs = [COLORREF + file for file in os.listdir(COLORREF)]
+    treepath_graphs = [TREEPATHS + file for file in os.listdir(TREEPATHS)]
+    return branching_graphs + colorref_graphs + treepath_graphs
+
+
+def process_graph_files(graph_files: List[str]):
+    for file in graph_files:
+        graphs = get_graphs_from_file(file)
+        isomorphs, iso_time, automorphs, auto_time = process_graphs(graphs)
+        result_string = stringify_result(file, isomorphs, iso_time, automorphs, auto_time)
+        output_result(result_string)
+
+
+def get_graphs_from_file(file: str) -> List[Graph]:
+    with open(file) as f:
+        L = load_graph(f, read_list=True)
+    return L[0]
+
+
+def process_graphs(graphs: List[Graph]) -> Tuple[List[List[Graph]], float, List[int], float]:
+    output_result(create_title_string("ISOMORPHISMS"))
+    iso_start = time.time()
+    preprocessed_graphs = preprocess_isomorphisms(graphs)
+    isomorphs = calculate_isomorphisms(preprocessed_graphs)
+    iso_time = time.time() - iso_start
+
+    output_result(create_title_string("AUTOMORPHISMS"))
```

Comment:
Misschien dat we hier straks nog een boolean in willen hebben zodat we alleen het GI probleem oplossen (de sets van isomorphism).